README.md

```diff
@@ -201,6 +201,69 @@ def test_render_component_for_tablet
 end
 ```
 
+### Previewing Components
+Action View Component previews provide a way to see how components look by visiting a special URL that renders them.
+In the previous example, the preview class for `TestComponent` should be named `TestComponentPreview` and located in `test/components/previews/test_component_preview.rb`.
+To see the preview of the component with a given title, implement a method that returns the locals (as a hash) to be passed to the component.
+The method can be named after what you want to preview given the locals you use, and you can define many examples (methods) as you want:
+
+```ruby
+# test/components/previews/test_component_preview.rb
+
+class TestComponentPreview < ActionView::Component::Preview
+  def with_default_title
+    {
+      title: "Test component default"
+    }
+  end
+
+  def with_long_title
+    {
+      title: "This is a really long title to see how the component renders this"
+    }
+  end
+end
+```
+
+Then the previews will be available in <http://localhost:3000/rails/components/test_component/with_default_title>
+and <http://localhost:3000/rails/components/test_component/with_long_title>.
+
+To specify the stylesheets and javascript used by your previews you must create a dummy layout file in `test/components/previews/layouts/`.
```

Comment:
Hmm. I'm wondering if it might be possible to reuse at least part of the application layout here. I'm worried that having a separate layout file for this would cause the previews to be inaccurate.

Reply:
At the very least, perhaps we could define a default layout that includes the default application styles and JS? That way developers could override that layout if necessary, but wouldn't have to create one. #conventionoverconfiguration

Reply:
Yep, great idea! 

So to sum up:
1. Create a default layout that will be used for rendering the preview examples
2. Developers should be able to override this layout

Reply:
Instead of creating a default layout inside the gem, do you think it's fine if we just use the application's layout? https://github.com/github/actionview-component/pull/96/commits/7df33aff785906097ad5fbaa0efa4dac381adac0#diff-67f888315429841b717b9aeef54e71a6R48

Developers will be able to set any other layout they want, but layouts must live inside their app. Makes sense?

Reply:
What do you think if we actually use the application's layout? https://github.com/github/actionview-component/pull/96/commits/7df33aff785906097ad5fbaa0efa4dac381adac0#diff-67f888315429841b717b9aeef54e71a6R48 So that way we don't need to create any layout on the gem's side. 

Developers will still be able to override this layout, using any of their application's layouts.

Reply:
I think it's worth a shot! If we go that route, do we still need to us iframes?

Reply:
Yes, because iframes will contain the component html inside the application's layout. And those iframes will live inside a view that uses the rails' application layout.

Does it make sense to you? Or do you have any other approach in mind?

Reply:
I guess I'm curious as to whether we need the iframes at all if we are just using the application layout.

I guess that would only work if you weren't using a different layout. Perhaps we could remove the custom layout functionality for now and see if anyone asks for it?

Reply:
Oh maybe we are not talking about the same thing. I forgot to update the readme. 

There are no custom layouts for component previews, developers can only use their application's layouts. By default previews will use the `app/views/layouts/application.html.erb` layout. 

But since the previews are displayed from a rails namespace, they are served [within a layout that comes from rails itself](https://github.com/rails/rails/blob/master/railties/lib/rails/templates/layouts/application.html.erb).

That's why we use iframes, to avoid having that layout when rendering the component. And also to avoid using the application's layout in the `rails/components` and `rails/components/:component_name` routes

I'll update the readme tonight

Reply:
Gotcha. Based on my comment below about staying consistent with ActionMailer::Preview, I think this is just fine for now. 😄 👍